cdap-docs/reference-manual/source/http-restful-api/http-restful-api-v3/metrics.rst

```diff
@@ -444,12 +246,356 @@ These metrics are available in the Datasets context:
 
    * - Datasets Metric
      - Description
-   * - ``store.bytes``
+   * - ``system.store.bytes``
      - Number of bytes written
-   * - ``store.ops``
+   * - ``system.store.ops``
      - Operations (reads and writes) performed
-   * - ``store.reads``
+   * - ``system.store.reads``
      - Read operations performed
-   * - ``store.writes``
+   * - ``system.store.writes``
      - Write operations performed
 
+
+Searches and Queries
+====================
+
+The process of retrieving a metric involves these steps:
+
+1. Obtain (usually through a search) the correct context for a metric;
+#. Obtain (usually through a search of the context) the available metrics;
+#. Querying for a specific metric, supplying the context and any parameters.
+
+
+Search for Contexts
+===================
+
+To search for the available contexts, perform an HTTP request::
+
+  POST <base-url>/metrics/search?target=childContext[&context=<context>]
+
+The optional ``<context>`` defines a metrics context to search within. If it is not
+provided, the search is performed across all data. The available contexts that are returned
+can be used to query for a lower-level of contexts.
+
+You can also define the query to search in a given context across all values of one or
+more tags provided in the context by specifying ``*`` as a value for a tag. See the
+examples below for its use.
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``<context>`` (Optional)
+     - Metrics context to search within. If not provided, the search is provided across
+       all contexts.
+       
+.. rubric:: Examples
+
+.. list-table::
+   :widths: 20 80
+   :stub-columns: 1
+
+   * - HTTP Method
+     - ``POST <base-url>/metrics/search?target=childContext``
+   * - Returns
+     - ``[ "namespace.default", "namespace.system" ]``
+   * - Description
+     - Returns all first-level contexts; in this case, two namespaces.
+   * - 
+     - 
+   * - HTTP Method
+     - ``POST <base-url>/metrics/search?target=childContext&context=namespace.default``
+   * - Returns
+     - ``[ "namespace.default.app.HelloWorld", "namespace.default.app.PurchaseHistory", "namespace.default.dataset.purchases", 
+       "namespace.default.dataset.whom", "namespace.default.stream.purchaseStream", "namespace.default.stream.who" ]``
+   * - Description
+     - Returns all child contexts of the given parent context; in this case, all entities in the default namespace.
+   * - 
+     - 
+   * - HTTP Method
+     - ``POST <base-url>/metrics/search?target=childContext&context=namespace.default.app.PurchaseHistory.flow.PurchaseFlow.dataset.*.run.*``
+   * - Returns
+     - ``[ "namespace.default.app.PurchaseHistory.flow.PurchaseFlow.dataset.*.run.*.flowlet.collector", 
```

Comment:
it does not return stars. You can use star to query, but it will not return one..